funcframework/framework_test.go

```diff
@@ -441,3 +443,27 @@ func TestCloudEventFunction(t *testing.T) {
 		}
 	}
 }
+
+func TestDeclarativeFunction(t *testing.T) {
+	funcName := "httpfunc"
+	os.Setenv("FUNCTION_TARGET", funcName)
+
+	h := http.NewServeMux()
+	overrideHandlerWithRegisteredFunctions(h)
+
+	// register functions
+	HTTP(funcName, func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, "Hello World!")
+	})
+
+	if _, ok := registry.GetRegisteredFunction(funcName); !ok {
+		t.Fatalf("could not get registered function: %s", funcName)
+	}
+
+	srv := httptest.NewServer(h)
+	defer srv.Close()
+
+	if _, err := http.Get(srv.URL); err != nil {
+		t.Fatalf("error: %s", err)
```

Comment:
[nit] error should describe what errored

Reply:
Thanks, I've added a better error message.